presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -505,49 +514,23 @@ public Void visitOperatorContext(OperatorContext operatorContext, AtomicLong rem
-            List<PrestoSparkShuffleInput> remoteSourceRowInputs = new ArrayList<>();
-            List<java.util.Iterator<PrestoSparkSerializedPage>> remoteSourcePageInputs = new ArrayList<>();
-            List<List<?>> broadcastInputsList = new ArrayList<>();
-            for (PlanFragmentId sourceFragmentId : remoteSource.getSourceFragmentIds()) {
-                Iterator<Tuple2<MutablePartitionId, PrestoSparkMutableRow>> shuffleInput = inputs.getShuffleInputs().get(sourceFragmentId.toString());
-                Broadcast<?> broadcastInput = inputs.getBroadcastInputs().get(sourceFragmentId.toString());
-                List<PrestoSparkSerializedPage> inMemoryInput = inputs.getInMemoryInputs().get(sourceFragmentId.toString());
-
-                if (shuffleInput != null) {
-                    checkArgument(broadcastInput == null, "single remote source is not expected to accept different kind of inputs");
-                    checkArgument(inMemoryInput == null, "single remote source is not expected to accept different kind of inputs");
-                    remoteSourceRowInputs.add(new PrestoSparkShuffleInput(sourceFragmentId.getId(), shuffleInput));
-                    continue;
-                }
-
-                if (broadcastInput != null) {
-                    checkArgument(inMemoryInput == null, "single remote source is not expected to accept different kind of inputs");
-                    // TODO: Enable NullifyingIterator once migrated to one task per JVM model
-                    // NullifyingIterator removes element from the list upon return
-                    // This allows GC to gradually reclaim memory
-                    // remoteSourcePageInputs.add(getNullifyingIterator(broadcastInput.value()));
-                    broadcastInputsList.add((List<?>) broadcastInput.value());
-                    continue;
-                }
+        ImmutableMap.Builder<PlanNodeId, PrestoSparkShuffleReadInfo> shuffleReadInfos = ImmutableMap.builder();
 
-                if (inMemoryInput != null) {
-                    // for in-memory inputs pages can be released incrementally to save memory
-                    remoteSourcePageInputs.add(getNullifyingIterator(inMemoryInput));
-                    continue;
-                }
-
-                throw new IllegalArgumentException("Input not found for sourceFragmentId: " + sourceFragmentId);
-            }
-            if (!remoteSourceRowInputs.isEmpty()) {
-                shuffleInputs.put(remoteSource.getId(), remoteSourceRowInputs);
-            }
-            if (!remoteSourcePageInputs.isEmpty()) {
-                pageInputs.put(remoteSource.getId(), remoteSourcePageInputs);
-            }
-            if (!broadcastInputsList.isEmpty()) {
-                broadcastInputs.put(remoteSource.getId(), broadcastInputsList);
-            }
+        List<TaskSource> taskSources = getTaskSources(serializedTaskSources);
```

Comment:
nit: Declare the classes here and do not initialize them. Set them to the right value in the if/else path. If they need to be populated and if you don't compiler will warn you. Giving them default values, and if you forgot to set them compiler won't warn you.